pyfr/solvers/base/elements.py

```diff
@@ -205,6 +199,11 @@ def _make_sliced_kernel(self, kseq):
     def set_backend(self, backend, nscalupts, nonce, linoff):
         self._be = backend
 
+        # If we are doing gradient fusion
+        be_pref = self._be.pref_grad_fusion
+        self.grad_fusion = (self.cfg.getbool('solver', 'grad-fusion', be_pref)
```

Comment:
Don't put a default here (in fact I would not expose it through the config file) since a default will prevent the option from changing on restart with a different backend.